src/app/apiTest/page.tsx

```diff
@@ -13,169 +16,94 @@ export default function Login() {
   const [endTime, setEndTime] = useState('');
   const [color, setColor] = useState('');
   const [todoId, setTodoId] = useState('');
-  // commit위한 주석
-  const createUser = async (event: React.FormEvent) => {
-    event.preventDefault();
 
-    const response = await fetch('/api/user/create', {
-      method: 'POST',
-      headers: {
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify({ email, password, nickname }),
-    });
+  const { refetch: refetchGetUserId } = useQuery({
+    queryKey: ['userId', email],
+    queryFn: () => getUserIdByEmail(email),
+    enabled: false,
+  });
```

Comment:
refetch라는 속성으로 네이밍을 한 이유가 따로 있을까요?
얘는 데이터를 가져온다는 성격이 강해 보통은 data로 많이 쓰는 것 같아서요

데이터를 불러오는 코드는 독립적인 파일로 분리가 되어있는게 더 낫지 않을까? 라는 생각이 들기도 합니다.

추가적으로 refetchGetUserId에서는 email을 인자로 사용하고 있는데 따로 인자로 받는 것이 존재하지 않습니다.
이는 아마 같은 파일에 존재하기 때문에라고 생각이 되는데요. api를 호출하는 hook이 분리 된다면 이는 인자를 받는 것이 필요할 것입니다.

```suggestion
  const useGetUserId = (emailInput: string) => {
    const { data: getUserId } = useQuery({
      queryKey: ['userId', emailInput],
      queryFn: () => getUserIdByEmail(emailInput),
    });
    return getUserId;
  };
```

enabled 속성 또한 의문이 듭니다.
해당 api는 submit이라는 버튼이 눌렸을 때 터지는 api라고 생각이 됩니다.
이것이 useGetUserId라는 함수로 분리가 된다면, 이는 onSubmit이 호출될 때만 부를 수 있을 것 같습니다.
그리고 이러한 방식이 해당 api를 다른 곳에서 쓸 때에서도 조금 더 이점이 있다고 생각이 됩니다.
어떻게 생각하시나요? 

아래에 있는 api호출들도 따로 파일을 분리하는 것이 낫다고 생각이 됩니다.
따라서 이것이 페이지가 렌더링 될때, 자연스럽게 되야하는 것인지, 어떠한 이벤트가 터지며 호출이 되어야 하는 것인지를 고려하여 enabled속성을 사용했으면 좋겠다라는 생각입니다!

Reply:
정성스런 코드리뷰 감사합니다:)  말씀하신 부분 저도 모두 동의합니다. 테스팅 페이지라 단순 작동에만 목적을 두게 되어 깔끔하지 못한 코드가 되었던 것 같습니다. 다만 현재 todo컴포넌트 팀과 연동이 필요한 상황이라, 우선적으로 marge 후 말씀하신 부분 적용하도록 할게요!